### ISSUE TITLE
Running jdftx on Frontier?

### ISSUE THREAD

**Issue by ColinBundschu** [shankar1729/jdftx#374]
Is it possible to run jdftx on Frontier? I was just awarded a significant amount of time and would like to make it work if possible. It uses AMD Instinct MI250X GPUs. I am prepared to do a substantial amount of work to make it compile and run.
https://www.olcf.ornl.gov/olcf-resources/compute-systems/frontier/

**Comment by shankar1729** [comment]
JDFTx only supports GPUs through CUDA, and was not ported to HIP, OpenACC etc. I believe there are now technqiues to automatically convert CUDA to HIP code etc., so it may be possible.

This is going to be a lot of work though, as many of the CUDA kernels, especially for the ultrasoft pseudopotentials, are quite non-trivial. The auto-conversion may therefore fail, and may require rewriting some kernels to support HIP. Either way, this is a guess - I have not worked with AMD GPUs at all.

**Comment by ColinBundschu** [comment]
It seems worthwhile to try and tackle AMD GPUs at some point though, no? They are becoming more common, and as the ecosystem matures eventually I imagine you will want to support both. I am hoping you agree there is no time like the present! :)

**Comment by shankar1729** [comment]
Yes, it will definitely be useful if you can get the port to work! I can help with debugging if you have it done part of the way, but I won't have much bandwidth overall for this.

**Comment by ColinBundschu** [comment]
Ok, I will try to tackle this then. Right now my plan is to use `cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=ON` and then `hipify-clang` on the output to try and automatically convert the whole project and debug from there. If you know of any specific places I might run into issues in advance, please let me know. Thanks for the help!

**Comment by mmarchywka** [comment]
as I was searching for ways to use my alleged AMD GPU, 
I too came across hipify but was not sure what has been
done. 

Thanks.
